index.js

```diff
@@ -0,0 +1,46 @@
+require("dotenv").config();
+const functions = require("firebase-functions");
+const ax = require("axios");
+
+const axios = ax.create({
+  baseURL: `${process.env.API_URL}`,
+  timeout: 1500,
+  headers: {
+    "Content-Type": "application/json",
+    Accept: "application/json",
+  },
+});
+
+axios.interceptors.response.use(
+  (response) => response,
+  (error) => Promise.reject(error)
+);
+
+exports.seo = functions.https.onRequest(async (req, res) => {
+  const websiteURL = `${req.headers["x-forwarded-proto"]}://${req.headers["x-forwarded-host"]}`;
+
+  const { data: html } = await ax.get(websiteURL);
+
+  const { data: meta } = await axios.post(`/seo`, {
+    url: websiteURL + req.url,
+  });
+
+  html = getFilledWithTags(html, meta);
+  res.send(html);
+});
+
+const getFilledWithTags = (html, { title, desc, image }) => {
+  return html.replace(
+    `<head>`,
+    `<head>
+          <meta property="og:title" content="${title}" />
+          <meta property="og:description" content="${desc}" />
+          <meta property="og:image" content=${image} />
```

Comment:
You need to replace all double and single quotes with (`) otherwise the title or description can break the page if they contain it
```
title.replace(/["|']/g, '`')
```

also if there are no image in html you will see `content="https://combat-sport.clubundefined"` looks ugly

```
content="${image || '/'}"
```

Reply:
Good point
You are now very experienced in this.

Reply:
I just eat a dog on aatlas 🐕